sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTestBase.java

```diff
@@ -1519,7 +1519,25 @@ TextAnalyticsClientBuilder getTextAnalyticsAsyncClientBuilder(HttpClient httpCli
         TextAnalyticsServiceVersion serviceVersion, boolean isStaticResource) {
         TextAnalyticsClientBuilder builder = new TextAnalyticsClientBuilder()
             .endpoint(getEndpoint(isStaticResource))
-            .httpClient(httpClient == null ? interceptorManager.getPlaybackClient() : httpClient)
+            .httpClient(httpClient)
+            .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
+            .serviceVersion(serviceVersion);
+        if (getTestMode() == TestMode.RECORD) {
+            builder.addPolicy(interceptorManager.getRecordPolicy());
+        }
+        if (getTestMode() == TestMode.PLAYBACK) {
+            builder.credential(new AzureKeyCredential(FAKE_API_KEY));
+        } else {
+            builder.credential(new AzureKeyCredential(getApiKey(isStaticResource)));
+        }
+        return builder;
+    }
+
+    TextAnalyticsClientBuilder getTextAnalyticsClientBuilder(HttpClient httpClient,
```

Comment:
how is this method different from getTextAnalyticsAsyncClientBuilder ?

Reply:
Good catch! They are the same. Will merge them into one util method.